rootfs/apply.go

```diff
@@ -39,8 +39,9 @@ import (
 // used to transport that tar. The blob descriptor may or may not describe
 // a compressed object.
 type Layer struct {
-	Diff ocispec.Descriptor
-	Blob ocispec.Descriptor
+	Diff         ocispec.Descriptor
+	Blob         ocispec.Descriptor
+	DcParameters map[string][][]byte
```

Comment:
Do we need `DcParameters` here since we are passing around the `diff.Applier` object? We could initialize it in `Unpack` with the `WithDcParameters` option? And add the application of the config in `ApplyLayers`?

Reply:
Going backwards: The layer.DcParameters are used in rootfs/apply.go like this:

```
        desc, err = a.Apply(ctx, layer.Blob, mounts, diff.WithDcParameters(layer.DcParameters))
```

ApplyLayers takes a diff.Applier as parameter:

```
func ApplyLayers(ctx context.Context, layers []Layer, sn snapshots.Snapshotter, a diff.Applier)
```
We are calling it from image.go like this:

```
               unpacked, err = rootfs.ApplyLayer(ctx, layer, chain, sn, a)
```

'a' is created from this: `a  = i.client.DiffService()`

If we initialize the DiffService with the `DcParameters` then it will either have to carry some options around or the DcParameters directly. But do the `DcParameters` belong there? I find the Layer to be the better place...